packages/components/src/icon/index.tsx

```diff
@@ -44,9 +47,10 @@ type AdditionalProps< T > = T extends ComponentType< infer U >
 	? SVGProps< SVGSVGElement >
 	: {};
 
-export type Props< P > = BaseProps< P > & AdditionalProps< IconType< P > >;
+export type Props< P extends {} > = BaseProps< P > &
+	AdditionalProps< IconType< P > >;
 
-function Icon< P >( {
+function Icon< P extends {} >( {
```

Comment:
These `<Icon>` type changes are related to this breaking change in 4.8: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-8.html#unconstrained-generics-no-longer-assignable-to-

Reply:
would changing this to 

```suggestion
function Icon< P = {} >( {
```

remove the need for all of the other related changes in the components package?

Reply:
I'm skeptical whether we even need the generic. `React.ComponentType` is `< P = {} >` by default, and none of our usages are actually passing in explicit props types. I can't really imagine a situation where someone would want/need to do so, either.

I'd be ok with removing all the generics here for now 👉 https://github.com/WordPress/gutenberg/commit/c70c1d6884a291376a00abaa3ebafaa820079148

Reply:
Great suggestions! I removed the generic in [a28ad60](https://github.com/WordPress/gutenberg/pull/48299/commits/a28ad60d96e8b40d3cc9afa3303e0bf94632dd82).